### PR TITLE
Add nasm, dav1d, aom, gd, libavif and libxpm

### PR DESCRIPTION
**Pull request by rawlingsj**
Remaining dependencies needed for Nginx

Signed-off-by: James Rawlings <jrawlings@chainguard.dev>

### CODE REVIEW

**Comment by rawlingsj** [comment]
Needs...

https://github.com/wolfi-dev/os/pull/321
https://github.com/wolfi-dev/os/pull/323
https://github.com/wolfi-dev/os/pull/327
https://github.com/wolfi-dev/os/pull/328

**Comment by dlorenc** [comment]
Nice! I punted on nasm since yasm seems to be a fine substitute, but it's probably worth having both for compatibility. I'd suggest trying to use yasm first though.

**Comment by rawlingsj** [comment]
> Nice! I punted on nasm since yasm seems to be a fine substitute, but it's probably worth having both for compatibility. I'd suggest trying to use yasm first though.

Ok yeah good call!  Will do.

**Comment by rawlingsj** [comment]
I'm not yet able to test locally if we can only use yasm.  Recently started getting some build errors which the GHA on this PR is failing with too.

Not too sure what's up yet.  There was a python upgrade recently so possibly related?  I [noticed when there's a go upgrade](https://github.com/wolfi-dev/os/pull/176/files) we do an epoch update for downstream dependencies, perhaps python is the same?

Build error:
```
  File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 563, in from_name
    return next(cls.discover(name=name))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
StopIteration

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/bin/meson", line 33, in <module>
    sys.exit(load_entry_point('meson==0.64.1', 'console_scripts', 'meson')())
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/bin/meson", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
                       ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 981, in distribution
    return Distribution.from_name(distribution_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/importlib/metadata/__init__.py", line 565, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: No package metadata was found for meson
```

**Comment by dlorenc** [comment]
> Not too sure what's up yet. There was a python upgrade recently so possibly related? I [noticed when there's a go upgrade](https://github.com/wolfi-dev/os/pull/176/files) we do an epoch update for downstream dependencies, perhaps python is the same?

I thought with go the issue was that we had to do that to get them all to rebuild with go, which we wanted to do because there was a vulnerability in go itself. My (probably naive) understanding is that since python isn't "built", it's closer to dynamic linking - we don't need to bump every package that uses glibc when we update glibc.

**Comment by rawlingsj** [comment]
Yeah that makes sense. I was having a very wild guess as not quite sure what's causing the build problem.

**Comment by dlorenc** [comment]
> Yeah that makes sense. I was having a very wild guess as not quite sure what's causing the build problem.

it's definitely possible some of these libraries don't work on the new version of python though...

**Comment by dlorenc** [comment]
I think this might help: https://github.com/wolfi-dev/os/pull/335

**Review by dlorenc — DISMISSED** [review]

**Comment by rawlingsj** [comment]
I "think" the last CI failure was because the release pipeline didn't complete properly because of two merges in close succession.  Going to retrigger a couple of GHA's.  They'll be more robust over time.

**Comment by rawlingsj** [comment]
Ok got past the python related issues but latest CI failure is related to thw `nasm` change 

```
2022/12/30 22:02:28 melange (dav1d/x86_64): Compiler for C supports arguments -mpreferred-stack-boundary=6: YES 
2022/12/30 22:02:28 melange (dav1d/x86_64): Fetching value of define "__AVX__" :  
2022/12/30 22:02:28 melange (dav1d/x86_64): Program nasm found: NO
2022/12/30 22:02:28 melange (dav1d/x86_64): 
2022/12/30 22:02:28 melange (dav1d/x86_64): meson.build:412:4: ERROR: Program 'nasm' not found or not executable
```

Think I'm going to add nasm back in and see if that helps

**Comment by dlorenc** [comment]
Ah weird, I thought it worked for me. Oh well, sorry!

**Comment by rawlingsj** [comment]
No worries!  It was definitely worth a try and TBH I'm going to double check in the AM but wanted to see if this goes green.

File: gd.yaml
Comment (dlorenc) [inline gd.yaml]: Needs to change to libjpeg-dev

File: libavif.yaml
Comment (dlorenc) [inline libavif.yaml]: And here

**Comment by rawlingsj** [comment]
yay - green!

**Review by dlorenc — APPROVED** [review]

File: aom.yaml
Comment (puerco) [inline aom.yaml:12]: Mmh this package has a special license to control how patents are used. In the SBOM you can add a custom license ref, but I'm not sure how to handle it in the apk, which will then flow back to the SBOM but without the context.

I think we will need to do an audit of all licensing and copyright entries in the packages to ensure we are keeping the notices and expressions correct.

**Review by puerco — APPROVED** [review]
OK, LGTM. Just one note about the license in one of them :point_down:

File: aom.yaml
Comment (rawlingsj) [reply aom.yaml:12]: Sounds good to me!  Raised an issue to track https://github.com/wolfi-dev/os/issues/340

Comment (rawlingsj) [reply aom.yaml:12]: @puerco are you thinking we need to resolve this before merging or would the audit be enough as a followup?

Comment (dlorenc) [reply aom.yaml:12]: I think we can clean it up after.

Comment (puerco) [reply aom.yaml:12]: @rawlingsj no I dind't mean to hold this one. I'm sure we have a bunch of inconsistent license entries, so one more will not change that. But we need to check them soon as the number of packages grows.

**Pull request merged**
